### ISSUE TITLE
Incorrect error diagnosis for one-sided t-tests

### ISSUE THREAD

**Issue by crsh** [MicheleNuijten/statcheck#17]
I think I found a bug in how `statcheck()` diagnoses errors in one-sided t-tests. Specifically, when the sample mean is in the "wrong" tail it is inappropriate to calculate the one-tailed *p*-value by halfing the two-sided *p*-value as is done by `statcheck()`.

```r
mean(sleep$extra[sleep$group == 1])
# [1] 0.75
mean(sleep$extra[sleep$group == 2])
# [1] 2.33

t.test(
  sleep$extra[sleep$group == 1]
  , sleep$extra[sleep$group == 2]
  , var.equal = T
)
# Two Sample t-test
# 
# data:  sleep$extra[sleep$group == 1] and sleep$extra[sleep$group == 2]
# t = -1.8608, df = 18, p-value = 0.07919
# alternative hypothesis: true difference in means is not equal to 0
# 95 percent confidence interval:
#   -3.363874  0.203874
# sample estimates:
#   mean of x mean of y 
# 0.75      2.33 

t.test(
  sleep$extra[sleep$group == 1]
  , sleep$extra[sleep$group == 2]
  , alternative = "greater"
  , var.equal = T
)
# 	Two Sample t-test
# 
# data:  sleep$extra[sleep$group == 1] and sleep$extra[sleep$group == 2]
# t = -1.8608, df = 18, p-value = 0.9604
# alternative hypothesis: true difference in means is greater than 0
# 95 percent confidence interval:
#  -3.052378       Inf
# sample estimates:
# mean of x mean of y 
#      0.75      2.33 
```

Hence, `statcheck` incorrectly indicates erroneous reporting, whereas the, in this case, incorrect *p*-value is deemed correct.

```r
statcheck:::statcheck("t(18) = -1.86, p = 0.960", OneTailedTests = TRUE)[, c("Reported.P.Value", "Computed", "Error", "OneTail")]
# Reported.P.Value   Computed Error OneTail
# 1           0.96 0.03965356  TRUE   FALSE

statcheck:::statcheck("t(18) = -1.86, p = 0.039", OneTailedTests = TRUE)[, c("Reported.P.Value", "Computed", "Error", "OneTail")]
# Reported.P.Value   Computed Error OneTail
# 1          0.039 0.03965356 FALSE   FALSE
```

Since `statcheck()` can't know what the tested hypothesis is, it should probably always consider both possibilities and err on the side of caution?

**Comment by MicheleNuijten** [comment]
Hi Frederik,

Thanks for your comment!
You're absolutely right that statcheck doesn't take into account the direction of the hypothesis. We included the automated one-tailed test detection to avoid flagging a ton of stats in which the detected inconsistency was easy to explain. However: also in these cases, the fact still is that these results are reported inconsistently, and that the correct way to report a one-tailed test is to adjust your alpha.

That's why for now I won't update this feature in the current version, but I will definitely put it on the list for statcheck 2.0!
